app/sidekiq/form526_status_polling_job.rb

```diff
@@ -34,33 +37,45 @@ def api_to_poll
   end
 
   def submissions
-    @submissions ||= Form526Submission.pending_backup_submissions
+    @submissions ||= if paranoid
+                       Form526Submission.paranoid_success_type
+                     else
+                       Form526Submission.pending_backup_submissions
+                     end
   end
 
   def handle_response(response)
     response.body['data']&.each do |submission|
       status = submission.dig('attributes', 'status')
       form_submission = Form526Submission.find_by(backup_submitted_claim_id: submission['id'])
 
-      if %w[error expired].include? status
-        log_result('failure')
-        form_submission.rejected!
-      elsif status == 'success'
-        Rails.logger.info('Form526StatusPollingJob', status: 'pending', submission_id: form_submission.id)
-      elsif status == 'vbms'
-        log_result('success')
-        form_submission.accepted!
-      else
-        Rails.logger.warn(
-          'Unknown status returned from Benefits Intake API for 526 submission',
-          status:,
-          submission_id: form_submission.id
-        )
-      end
+      handle_submission(status, form_submission)
       @total_handled += 1
     end
   end
 
+  def handle_submission(status, form_submission)
+    if %w[error expired].include? status
+      log_result('failure')
+      form_submission.rejected!
+    elsif status == 'vbms'
+      log_result('true success')
+      form_submission.accepted!
+    elsif status == 'success' && !paranoid
+      log_result('paranoid success')
+      form_submission.paranoid_success!
```

Comment:
I'm confused here. Isn't 'paranoid success' an indication of 'success' and `paranoid` is **true**, not if `paranoid` is **false**? Maybe I'm misinterpreting the evaluation.

Reply:
yeah it's weird... here's my thought process

When we run `paranoid: true` we are scoping our poll to submissions that are already set to `paranoid_success`. Therefor, if we are at this point during a `paranoid: true` run, and `status == 'success'`, then there is no reason to call `paranoid_success!`, simply because it would be redundant. 

However if we are running `paranoid: false` and return `status == 'success'`, then we are polling on a submission that currently is pending, but now should be marked paranoid_success. therefor, `status == 'success'` and `paranoid: false` indicates we should mark this a paranoid success.

Reply:
added

Reply:
That makes sense. Thanks for the explanation.